pkg/agent/waitfor.go

```diff
@@ -53,3 +53,36 @@ func WaitForBootstrapComplete(assetDir string) (*Cluster, error) {
 
 	return cluster, nil
 }
+
+// WaitForInstallComplete Waits for the cluster installation triggered by the
+// agent installer to be complete.
+func WaitForInstallComplete(assetDir string) (*Cluster, error) {
+
+	cluster, err := WaitForBootstrapComplete(assetDir)
+
+	if err != nil {
+		return cluster, errors.Wrap(err, "error occured during bootstrap process")
+	}
+
+	timeout := 90 * time.Minute
```

Comment:
90 minutes seems like a long time from bootstrap complete to cluster complete. IPI uses 60 minutes for baremetal and 40 otherwise.
In principle I think we should be faster, because once bootstrap is complete we have all of the hosts provisioned, but IPI still has to provision the workers.

It looks like we might be waiting for the console in here as well, so that's another 10 minutes, but really they should be separate timeouts because if the cluster doesn't come up in time we want to report that to the user, not continue waiting for another 10 minutes. Likewise, if the cluster is complete in less than its allotted maximum time, we still don't want to wait more than 10 minutes for the console.

Reply:
My experience with Assisted Installer tells me that there are lots of failures at the 1 and a half hour mark. The timeout should cancel early if it reaches a good state before then. I can drop it down for sure.